### DISCUSSION TITLE
Modelling switches in PGM

### DISCUSSION THREAD

**Discussion by PowerIsKnowledgeOrSomething** [PowerGridModel/.github#34] [Q&A] [answered]
Hello! I'm currently mapping power flow data from CSV files into PGM using Tabular Conversion, so far nodes, loads, lines, and transformers have been mapped, but I am having trouble modelling my switches. I've had a look through the information on component types in PGM, but cant seem to find a straight forward answer for switching types. Any advice would be much appreciated,
Sincerely,
Caleb

**Comment by mgovers** [comment]
Hi @PowerIsKnowledgeOrSomething,

It should be possible to use the `link` component in PGM for that. It has two sides that you can use to switch: `from_status` and `to_status` enabling both results in the `ON` state, while setting one to 0 yields a disconnected `OFF` switch state.

Hope this answers your question.

Martijn

**Comment by petersalemink95** [comment, marked as answer]
Hi @PowerIsKnowledgeOrSomething,

You can indeed use `link`s as @mgovers mentioned if you want to have specific switching components in PGM. However, for numerical stability it would be beneficial to have as little `link`s as possible. 

A better implementation would be to translate the switch status to the `from_status` or `to_status` of the specific branch (`link`, `line`, `transformer`). For example:

Your data: node1 - switch1 - line - switch2 - node2
PGM: node1 - line - node2

Where you can set the status of switch1 in `from_status` and the status of switch2 in `to_status` of your line (if `from_node` is node1 and `to_node` is node2; else, it will be the other way around)

**Reply by PowerIsKnowledgeOrSomething** [reply]
Ok thanks!
